File: src/main/java/gestion_restaurant/entity/Complement.java
package gestion_restaurant.entity;

import lombok.*;
import java.math.BigDecimal;
import java.time.Instant;

@Getter @Setter @NoArgsConstructor 
 @ToString
public class Complement {
    private Long id;
    private String nom;
    private BigDecimal prix;
    private String image;
    private ComplementType typeComplement;
    private Instant createdAt = Instant.now();
    private String imagePublicId;
    private String description;

    public Complement(Long id, String nom, BigDecimal prix, String image,
                      ComplementType typeComplement, Instant createdAt,
                      String imagePublicId, String description) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.image = image;
        this.typeComplement = typeComplement;
        this.createdAt = createdAt;
        this.imagePublicId = imagePublicId;
        this.description = description;
    }
}

File: Models/Complement.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace gestion_restaurant.Models
{
    [Table("complement")]
    public class Complement
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Column("nom")]
        public string Nom { get; set; }

        [Column("image")]
        public string Image { get; set; }

        [Column("prix")]
        public decimal Prix { get; set; }

        [Column("typecomplement")]
        public string TypeComplement { get; set; }

        [Column("image_public_id")]
        public string ImagePublicId { get; set; }

        [Column("description")]
        public string Description { get; set; }
    }
}
